photon-api/src/main/scala/com/linkedin/photon/ml/estimators/GameEstimator.scala

```diff
@@ -507,12 +508,32 @@ class GameEstimator(val sc: SparkContext, implicit val logger: Logger) extends P
             None
           }
 
-          val randomEffectDataset = RandomEffectDataset(
-            gameDataset,
-            reConfig,
-            rePartitioner,
-            existingModelKeysRddOpt,
-            StorageLevel.DISK_ONLY)
+          val randomEffectDataset = if(initialModelOpt.isDefined) {
+            val reModels = initialModelOpt.get.getModel(coordinateId).map {
+              case rem: RandomEffectModel =>
+                rem
+
+              case other =>
```

Comment:
If the `initialModelOpt` contains a fixed-effect model, this will throw an exception but we should let it be ignored instead.

Reply:
`coordinateId` will point to a random effect coordinate since the config is `RandomEffectDataConfiguration`